README.md
# skyl.fr

Personal portfolio site, a place to showcase what I did, interactive animations, music, drawings and so on.

#### 1er version (on branch p5):
A minimalist site made with Canvas API.

#### 2nd version (on branch vue):

Made with Vue, using p5 instance mode and sometimes iframe to host all the sketches within the SPA itself.

#### New Features :

'Play !' - Same as the first version, projects are categorised into 'work', 'music', 'drawings', and this time I added a 'play' page where users can quickly switch between differences projects and test them instantly. Also an introduction page for each project, with photos, descriptions, project contributors etc... 

'Read more' - a minimalist article page for project infos.

Dynamic imports and p5 instance mode to open new sketches on the fly, moving the .JS sketches files into the SPA itself makes it possible to add toolbox interface with Vue, easier to store data, also enables communication between sketches


## Project setup
```
yarn install
```

### Compiles and hot-reloads for development
```
yarn run serve
```

### Compiles and minifies for production
```
yarn run build
```

### Run your tests
```
yarn run test
```

### Lints and fixes files
```
yarn run lint
```

### Customize configuration
See [Configuration Reference](https://cli.vuejs.org/config/).


ChangeLog.md
## [2.12.1] - 2021-02-25
### Changed
- reaction-diffusion back to legacy version def9c8 ([2fc52f](https://github.com/sikaili/skyl.fr/commit/2fc52f))

## [2.12.0] - 2021-02-25
### Added
- rubrique photos ([9ed525](https://github.com/sikaili/skyl.fr/commit/9ed525))
- let ring ([81df7e](https://github.com/sikaili/skyl.fr/commit/81df7e))
### Fixed
- album navigation pb ([4d95d5](https://github.com/sikaili/skyl.fr/commit/4d95d5))
### Changed
- minify photos ([695467](https://github.com/sikaili/skyl.fr/commit/695467))

## [2.11.1] - 2020-12-12
### Fixed
- pages error (modif store) ([b5fb2e](https://github.com/sikaili/skyl.fr/commit/b5fb2e))
### Changed
- refacto google analytics ([637f02](https://github.com/sikaili/skyl.fr/commit/637f02))

## [2.11.0] - 2020-12-12
### Added
- new muisic, seed, image resize plugin ([5488d3](https://github.com/sikaili/skyl.fr/commit/5488d3))
- isNew option for projects ([a7c454](https://github.com/sikaili/skyl.fr/commit/a7c454))
- iframe hasmargin option ([26cecc](https://github.com/sikaili/skyl.fr/commit/26cecc))
- bubbles add font bounds, mouseConstraint, isActive status. ([03270e](https://github.com/sikaili/skyl.fr/commit/03270e))
### Changed
- password needed for staging env ([2ff7ba](https://github.com/sikaili/skyl.fr/commit/2ff7ba))
- show beta sketches only in dev mode ([5d32e8](https://github.com/sikaili/skyl.fr/commit/5d32e8))

## [2.10.0] - 2020-11-30
### Changed
- refacto seed.js, store.js, using sketches.json seed.json ([d9280b](https://github.com/sikaili/skyl.fr/commit/d9280b))
- optim bubbles ([071e53](https://github.com/sikaili/skyl.fr/commit/071e53))

## [2.9.12] - 2020-11-29
### Added
- nodejs create sketch automation ([17216e](https://github.com/sikaili/skyl.fr/commit/17216e))
### Fixed
- bubbles display ([bdb5ff](https://github.com/sikaili/skyl.fr/commit/bdb5ff))
### Changed
- node auto project optim and other minor changes ([5984e4](https://github.com/sikaili/skyl.fr/commit/5984e4))

## [2.9.11] - 2020-10-24
### Fixed
- offline on aws, set cache-control header with aws cli ([cbacac](https://github.com/sikaili/skyl.fr/commit/cbacac))
- webperf, undo lazyload canvas wrapper ([servic](https://github.com/sikaili/skyl.fr/commit/servic))
### Changed
- move google analytics to app ([0d5a3f](https://github.com/sikaili/skyl.fr/commit/0d5a3f))
- reaction-diffusion default settings ([8e4fe7](https://github.com/sikaili/skyl.fr/commit/8e4fe7))
- remove traffic, not comptible with new Tone.js version ([59e9b8](https://github.com/sikaili/skyl.fr/commit/59e9b8))

## [2.9.10] - 2020-10-10
### Changed
- update tone.js and so on ([aa4748](https://github.com/sikaili/skyl.fr/commit/aa4748))

## [2.9.9] - 2020-09-30
### Fixed
- tone js import ([d8ecc8](https://github.com/sikaili/skyl.fr/commit/d8ecc8))

## [2.9.8] - 2020-09-03
### Added
- menu item multi items:.apps in seeddata ([3d44a5](https://github.com/sikaili/skyl.fr/commit/3d44a5))
### Fixed
- ios chrome no canvas ([1aea56](https://github.com/sikaili/skyl.fr/commit/1aea56))
### Changed
- load p5 with loadscript ([10a2fc](https://github.com/sikaili/skyl.fr/commit/10a2fc))

## [2.9.7] - 2020-08-30
### Changed
- async import canvas wrapper, remove workbox rules for media file, fix player bug ([2e22a4](https://github.com/sikaili/skyl.fr/commit/2e22a4))

## [2.9.6] - 2020-07-28
### Fixed
- sw audio cache ([77c790](https://github.com/sikaili/skyl.fr/commit/77c790))
- songslug desktop (forgotten) ([b44223](https://github.com/sikaili/skyl.fr/commit/b44223))
### Changed
- load service work after page is loaded ([4e7aa9](https://github.com/sikaili/skyl.fr/commit/4e7aa9))
- next song songslug ([4b81e8](https://github.com/sikaili/skyl.fr/commit/4b81e8))

## [2.9.5] - 2020-07-27
### Fixed
- google analytics ([64fe97](https://github.com/sikaili/skyl.fr/commit/64fe97))
### Changed
- push songslug when change song, modified vue-aplayergit s ([47dfa4](https://github.com/sikaili/skyl.fr/commit/47dfa4))
- push songslug when change song, modified vue-aplayergit s ([4f332f](https://github.com/sikaili/skyl.fr/commit/4f332f))

## [2.9.4] - 2020-07-26
### Added
- gtag ([500773](https://github.com/sikaili/skyl.fr/commit/500773))
### Changed
- delete useless media file, vue-analytics ([8aa4be](https://github.com/sikaili/skyl.fr/commit/8aa4be))

## [2.9.3] - 2020-07-25
### Added
- loader ([e5f48c](https://github.com/sikaili/skyl.fr/commit/e5f48c))
- loader animation ([3e6d1b](https://github.com/sikaili/skyl.fr/commit/3e6d1b))
- multi songs, add song slug ([b980cb](https://github.com/sikaili/skyl.fr/commit/b980cb))
### Changed
- keep menu alive ([13b04c](https://github.com/sikaili/skyl.fr/commit/13b04c))

## [2.9.2] - 2020-07-24
### Fixed
- graphic ajustment ([73b9c0](https://github.com/sikaili/skyl.fr/commit/73b9c0))
- add skipwaiting ([db03d9](https://github.com/sikaili/skyl.fr/commit/db03d9))

## [2.9.1] - 2020-07-24
### Fixed
- service worker causing refresh ([958cb7](https://github.com/sikaili/skyl.fr/commit/958cb7))
### Changed
- disable sw debug ([77fd64](https://github.com/sikaili/skyl.fr/commit/77fd64))
- pause tone when on music page; go to menu item with router params ([95bd2e](https://github.com/sikaili/skyl.fr/commit/95bd2e))

## [2.9.0] - 2020-07-22
### Added
- music player ([ec22d9](https://github.com/sikaili/skyl.fr/commit/ec22d9))

## [2.8.7] - 2020-06-28
### Added
- article component, image component, date-fns, data, scss config & colors ([0da5f5](https://github.com/sikaili/skyl.fr/commit/0da5f5))
### Fixed
- article image css ([b9e333](https://github.com/sikaili/skyl.fr/commit/b9e333))

## [2.8.6] - 2020-06-28
### Changed
- reaction-diffusion:set text distance threshold ([224193](https://github.com/sikaili/skyl.fr/commit/224193))

## [2.8.5] - 2020-06-22
### Fixed
- reaction-diffusion sound ([05bbfd](https://github.com/sikaili/skyl.fr/commit/05bbfd))

## [2.8.4] - 2020-06-07
### Added
- reaction-diffusion sound ([7c9c6b](https://github.com/sikaili/skyl.fr/commit/7c9c6b))
### Changed

src/sw.js
// workbox.setConfig({ debug: true });
self.addEventListener('message', (event) => {
  console.log(event);
  if (event.data && event.data.type === 'SKIP_WAITING') {
    self.skipWaiting();
  }
});

self.__precacheManifest = [].concat(self.__precacheManifest || []);
const arr = self.__precacheManifest.filter((obj) => {
  if ((obj.url && obj.url.indexOf('/src/projects/player/sound/') === -1) && (obj.url.indexOf('img/') === -1 || obj.url.indexOf('img/covers') !== -1)) {
    return true;
  }
  return false;
});
console.log(arr);
self.__precacheManifest = [].concat(arr || []);

const precacheController = new workbox.precaching.PrecacheController('precache-list');
precacheController.addToCacheList(self.__precacheManifest);

self.addEventListener('install', (event) => {
  event.waitUntil(() => precacheController.install());
});

self.addEventListener('activate', (event) => {
  event.waitUntil(precacheController.activate());
});

// Make sure to return a specific response for all navigation requests.
// Since we have a SPA here, this should be index.html always.
// https://stackoverflow.com/questions/49963982/vue-router-history-mode-with-pwa-in-offline-mode
workbox.routing.registerNavigationRoute('/index.html');

workbox.routing.registerRoute(
  /^https:\/\/fonts\.googleapis\.com/,
  new workbox.strategies.CacheFirst({
    cacheName: 'google-fonts-stylesheets',
    plugins: [
      new workbox.expiration.Plugin({
        maxAgeSeconds: 60 * 60 * 24 * 365,
        maxEntries: 30,
      }),
    ],
  }),
);

workbox.routing.registerRoute(
  /^https:\/\/fonts\.gstatic\.com/,
  new workbox.strategies.CacheFirst({
    cacheName: 'google-fonts-webfonts',
    plugins: [
      new workbox.cacheableResponse.Plugin({
        statuses: [0, 200],
      }),
      new workbox.expiration.Plugin({
        maxAgeSeconds: 60 * 60 * 24 * 365,
        maxEntries: 30,
      }),
    ],
  }),
);

workbox.routing.registerRoute(
  /^https:\/\/cdnjs\.cloudflare\.com/,
  new workbox.strategies.CacheFirst({
    cacheName: 'ionicons',
    plugins: [
      new workbox.cacheableResponse.Plugin({
        statuses: [200],
      }),
      new workbox.expiration.Plugin({
        maxAgeSeconds: 60 * 60 * 24 * 365,
        maxEntries: 60,
      }),
    ],
  }),
);

workbox.routing.registerRoute(/tachyons.min.css$/,
  new workbox.strategies.CacheFirst({
    cacheName: 'tachyons',
    plugins: [
      new workbox.cacheableResponse.Plugin({
        statuses: [200],
      }),
      new workbox.expiration.Plugin({
        maxAgeSeconds: 60 * 60 * 24 * 365,
        maxEntries: 60,
      }),
    ],
  }));

workbox.routing.registerRoute(/p5.min.js$/,
  new workbox.strategies.CacheFirst({
    cacheName: 'p5',
    plugins: [
      new workbox.cacheableResponse.Plugin({
        statuses: [200],
      }),
      new workbox.expiration.Plugin({
        maxAgeSeconds: 60 * 60 * 24 * 365,
        maxEntries: 60,
      }),
    ],
  }));

workbox.routing.registerRoute(/\.(?:png|gif|jpg)$/,
  new workbox.strategies.CacheFirst({
    cacheName: 'image-cache',
    plugins: [
      new workbox.cacheableResponse.Plugin({
        statuses: [200],
      }),
      new workbox.rangeRequests.Plugin(),
      new workbox.expiration.Plugin({
        maxAgeSeconds: 60 * 60 * 24 * 30,
        maxEntries: 60,
      }),
    ],
  }));

workbox.routing.registerRoute(/\.(?:m4a)$/,
  new workbox.strategies.NetworkOnly({
    cacheName: 'm4a',
  }));

workbox.routing.registerRoute(
  ({ request }) => request.destination === 'script',
  new workbox.strategies.CacheFirst({
    cacheName: 'static-resources',
  }),
);

const { setDefaultHandler, setCatchHandler } = workbox.routing;
const { StaleWhileRevalidate } = workbox.strategies;

// Use a stale-while-revalidate strategy for all other requests.
setDefaultHandler(new StaleWhileRevalidate());

// This "catch" handler is triggered when any of the other routes fail to
// generate a response.
setCatchHandler(({ event }) => {
  // The FALLBACK_URL entries must be added to the cache ahead of time, either
  // via runtime or precaching. If they are precached, then call
  // `matchPrecache(FALLBACK_URL)` (from the `workbox-precaching` package)
  // to get the response from the correct cache.
  //
  // Use event, request, and url to figure out how to respond.
  // One approach would be to use request.destination, see
  // https://medium.com/dev-channel/service-worker-caching-strategies-based-on-request-types-57411dd7652c
  switch (event.request.destination) {
    case 'document':
      // If using precached URLs:
      // return matchPrecache(FALLBACK_HTML_URL);
      return precacheController.matchPrecache('index.html');
      break;
    case 'image':
      // If using precached URLs:
      // return matchPrecache(FALLBACK_IMAGE_URL);
      return precacheController.matchPrecache('/');
      break;
    case 'font':
      // If using precached URLs:
      // return matchPrecache(FALLBACK_FONT_URL);
      return precacheController.matchPrecache('/');
      break;
    default:
      return Response.error();
  }
});
